manifests/init.pp

```diff
@@ -243,68 +280,68 @@
-  $log_stderr                          = $::corosync::params::log_stderr,
-  $syslog_priority                     = $::corosync::params::syslog_priority,
-  $log_function_name                   = $::corosync::params::log_function_name,
-  $rrp_mode                            = $::corosync::params::rrp_mode,
-  $netmtu                              = $::corosync::params::netmtu,
-  $ttl                                 = $::corosync::params::ttl,
-  $vsftype                             = $::corosync::params::vsftype,
-  $package_corosync                    = $::corosync::params::package_corosync,
-  $package_crmsh                       = $::corosync::params::package_crmsh,
-  $package_pacemaker                   = $::corosync::params::package_pacemaker,
-  $package_pcs                         = $::corosync::params::package_pcs,
-  $packageopts_corosync                = $::corosync::params::package_install_options,
-  $packageopts_pacemaker               = $::corosync::params::package_install_options,
-  $packageopts_crmsh                   = $::corosync::params::package_install_options,
-  $packageopts_pcs                     = $::corosync::params::package_install_options,
-  $version_corosync                    = $::corosync::params::version_corosync,
-  $version_crmsh                       = $::corosync::params::version_crmsh,
-  $version_pacemaker                   = $::corosync::params::version_pacemaker,
-  $version_pcs                         = $::corosync::params::version_pcs,
-  $set_votequorum                      = $::corosync::params::set_votequorum,
-  $votequorum_expected_votes           = $::corosync::params::votequorum_expected_votes,
-  $quorum_members                      = ['localhost'],
-  $quorum_members_ids                  = undef,
-  $quorum_members_names                = undef,
-  $token                               = $::corosync::params::token,
-  $token_retransmits_before_loss_const = $::corosync::params::token_retransmits_before_loss_const,
-  $compatibility                       = $::corosync::params::compatibility,
-  $enable_corosync_service             = $::corosync::params::enable_corosync_service,
-  $enable_pacemaker_service            = $::corosync::params::enable_pacemaker_service,
-  $manage_pacemaker_service            = $::corosync::params::manage_pacemaker_service,
-  $enable_pcsd_service                 = $::corosync::params::enable_pcsd_service,
-  $manage_pcsd_service                 = false,
-  $cluster_name                        = $::corosync::params::cluster_name,
-  $join                                = $::corosync::params::join,
-  $consensus                           = $::corosync::params::consensus,
-  $max_messages                        = $::corosync::params::max_messages,
-  $test_corosync_config                = $::corosync::params::test_corosync_config,
+  Boolean $enable_secauth                                = $::corosync::params::enable_secauth,
```

Comment:
$enable_secauth allows a Boolean or 'on'/'off'. To be backward-compatible with user configuration, it would need a custom datatype. (Variant[Boolean, Enum['true', 'false']])

Reply:
I see. Data type of `$enable_secauth` is now `Variant[Boolean, Enum['on', 'off']]` (see fc70b0cd).

Reply:
I don't think it should be backwards compatible. If it supports Boolean, we should _only_ support Boolean.

Reply:
Actually, code support both: https://github.com/voxpupuli/puppet-corosync/blob/59b9dc0be7d5984fdd26245a718f60041fbd73a1/manifests/init.pp#L342

At the end, the code writes 'on' or 'off' in the corosync.conf.erb. In #401, @kBite proposal was to drop the Boolean support to keep only the 'on'/'off' flags. It has been asked to made this change in a separate PR, thus #419 can focus on puppet4 data types only.

Reply:
This _could_ be a breaking change for some which do use 'on' or 'off'.

Reply:
Yes, this is a massively breaking changeset already, so now is the time that we should nudge things to a more standard direction. IMHO, it should be either enum on / off _or_ Boolean (preferably the latter), but not both.

Reply:
I do agree, there can be only one. But I strongly believe it should be done in another PR to discuss this change without blocking migration to data types.

Reply:
That works for me.